### ISSUE TITLE
IDE0079 (remove unnecessary suppressions) does not work when EnforceCodeStyleInBuild is enabled

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#47942]
**Version Used**: VS2019 16.8 P3

**Steps to Reproduce**:

1. Create a net5.0 class library project
```csharp
#pragma warning disable UnknownId1 // IDE0079

[System.Diagnostics.CodeAnalysis.SuppressMessage("Category", "UnknownId2")]  // IDE0079
public class C { }
```
2. Verify the pragma and suppress message attribute gets faded with IDE0079
3. Either install the CodeStyle NuGet package from [here](https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet-tools&view=overview&package=Microsoft.CodeAnalysis.CSharp.CodeStyle&version=3.8.0-3.20460.2&protocolType=NuGet) or set [EnforceCodeStyleInBuild](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#enforcecodestyleinbuild) to `true`

**Expected Behavior**:
IDE0079 is reported

**Actual Behavior**:
IDE0079 is not reported

This is a special IDE-only analyzer that should not be exported in the CodeStyle NuGet package. We still want to retain the code in the shared analyzer layer to allow different analyzer hosts to be able to explicitly instantiate it and enforce it on command line.

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/47944
